File: hello.py
import libhello
import numpy as np
print("From libhello : ", libhello.Greet())

print(libhello.GetMat())

a = np.random.randn(3, 3)
print('----Print Matrix from python')
print(a)
#a = np.zeros((3, 3), dtype=np.float64)
print('----Print Matrix from C++')
libhello.PrintMat(a)

print('----Get Matrix from C++')
m = libhello.GetMat()
print(m)

File: hello.cpp
#include <iostream>
#include <Eigen/Dense>

using Eigen::Matrix3f;
using Eigen::Matrix3d;
using namespace std;

void PrintMat(const Matrix3d& m) {
  cout << "Matrix : " << m << endl;
}

Eigen::Matrix3f GetMat() {
  Matrix3f m;
  m << 1, 2, 3,
       4, 5, 6,
       7, 8, 9;
  return m;
}

char const* Greet() {
  return "Hello, world ! ";
}

